src/app/applications/components/application-resource-diff/application-resource-diff.tsx

```diff
@@ -16,12 +17,17 @@ export class ApplicationResourceDiff extends React.Component<ApplicationComponen
     }
 
     public render() {
-        let liveState = this.props.liveState || {};
-        if (this.state.hideDefaultedFields) {
-            liveState = this.removeDefaultedFields(this.props.targetState || {}, liveState);
+        let diff = JSON.parse(this.props.diff.diffs);
+        const liveState = this.props.liveState;
+        const diffPatcher = new DiffPatcher();
+        if (!this.state.hideDefaultedFields ) {
+            if (liveState !== null && liveState.hasOwnProperty('metadata') && liveState.metadata.hasOwnProperty('annotations')) {
```

Comment:
I feel like there is a better way of doing this. @alexmt can you confirm that?

Reply:
You can simplify it a little bit:

```
if (liveState && liveState.metadata && liveState.metadata.annotations) {
```

Reply:
fixed